AuthService/Data/DALUser.cs

```diff
@@ -0,0 +1,182 @@
+            var key = Encoding.ASCII.GetBytes(_config["AppSettings:Secret"]);
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(claims),
+
+                Expires = DateTime.UtcNow.AddHours(1),
+
+                SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key),
+                    SecurityAlgorithms.HmacSha256Signature),
+            };
+
+            var token = tokenHandler.CreateToken(tokenDescriptor);
+            var tokenString = tokenHandler.WriteToken(token);
+
+            user.Token = tokenString;
+
+            return tokenString;
+        }
+
+        public IEnumerable<ApplicationUser> GetAllUser()
+        {
+            return _um.Users.ToList();
+        }
+
+        public async Task<ApplicationUser> GetByUserId(string userId)
+        {
+            var result = await _um.FindByIdAsync(userId);
+
+            if (result == null) throw new Exception("User not found");
+
+            return result;
+        }
+
+        public async Task<IList<string>> GetRolesByUsername(string username)
+        {
+            var user = await _um.FindByNameAsync(username);
+
+            var roles = await _um.GetRolesAsync(user);
+
+            return roles;
+        }
+
+        public async Task<ApplicationUser> Insert(DtoUserRegister input)
+        {
+            try
+            {
+                var newUser = new ApplicationUser
+                {
+                    UserName = input.Username,
+                    IsBlocked = false,
+                };
+
+                var result = await _um.CreateAsync(newUser, input.Password);
+
+                if (!result.Succeeded) throw new Exception($"Failed to create user: {result.ToString()}");
+
+                if (input.Role == role.driver)
+                {
+                    var newDriver = new DtoDriver
+                    {
+                        FirstName = input.FirstName,
+                        LastName = input.LastName,
+                        BirthDate = input.BirthDate,
+                        Latitude = 0,
+                        Longitude = 0,
+                        Balance = 0,
+                        IsApprove = false,
+                        UserId = newUser.Id
+                    };
+
+                    // TODO Send InsertDriver Request to DriverService
+
+                    await _um.AddToRoleAsync(newUser, "Driver");
+                }
+                else if (input.Role == role.customer)
+                {
+                    var newCust = new DtoCustomer
+                    {
+                        FirstName = input.FirstName,
+                        LastName = input.LastName,
+                        BirthDate = input.BirthDate,
+                        Balance = 0,
+                        UserId = newUser.Id,
+                    };
+
+                    // TODO Send InsertCustomer Request to CustomerService
```

Comment:
Ditto, lupa implementasi

Reply:
Implemented in https://github.com/mystald/Bootcamp_Project_2/pull/14/commits/f40938a8c1146457ed4dd1c3c02a6eaad9d6bf0d